plugins/inventory/iocage.py

```diff
@@ -201,11 +201,10 @@ def get_inventory(self, path):
             try:
                 t_stdout = to_text(stdout, errors='surrogate_or_strict')
             except UnicodeError as e:
-                raise AnsibleError('Invalid (non unicode) input returned: %s' % to_native(e)) from e
+                raise AnsibleError(f'Invalid (non unicode) input returned: {e}') from e
 
         except Exception as e:
-            raise AnsibleParserError('Failed to parse %s: %s' %
-                                     (to_native(path), to_native(e))) from e
+            raise AnsibleParserError(f'Failed to parse {to_native(path)}: {e}') from e
```

Comment:
```suggestion
            raise AnsibleParserError(f'Failed to parse {path}: {e}') from e
```

Reply:
`path` could be a byte string, in which case the output would change from `Failed to parse /foo/` to `Failed to parse b'/foo/'`. (I didn't check whether it really could be a byte string, I only removed to_native/str/... from places where it was obviously safe...)